app/src/contexts/observationsTableContext.tsx

```diff
@@ -751,43 +899,177 @@ export const ObservationsTableContextProvider = (props: PropsWithChildren<Record
     ]
   );
 
-  const isLoading: boolean = useMemo(() => {
-    return !taxonomyCacheStatus.isInitialized || isLoadingObservationsData;
-  }, [isLoadingObservationsData, taxonomyCacheStatus.isInitialized]);
+  /**
+   * Compiles all measurement columns and their values for the given row.
+   *
+   * @param {ObservationRecord} row
+   * @return {*}
+   */
+  const _getMeasurementsToSave = useCallback(
+    (row: ObservationRecord) => {
+      const measurementDefinitions = measurementColumns.map((item) => item.measurement);
+
+      const qualitative: SubcountToSave['qualitative'] = [];
+      const quantitative: SubcountToSave['quantitative'] = [];
+
+      // For each measurement column in the data grid
+      for (const measurementDefinition of measurementDefinitions) {
+        // If the row has a non-null/non-undefined value for the measurement column
+        if (row[measurementDefinition.taxon_measurement_id]) {
+          // If the measurement column is a qualitative measurement, add it to the qualitative array
+          if (isQualitativeMeasurementTypeDefinition(measurementDefinition)) {
+            qualitative.push({
+              measurement_id: measurementDefinition.taxon_measurement_id,
+              measurement_option_id: row[measurementDefinition.taxon_measurement_id]
+            });
+          }
+
+          // If the measurement column is a quantitative measurement, add it to the quantitative array
+          if (isQuantitativeMeasurementTypeDefinition(measurementDefinition)) {
+            quantitative.push({
+              measurement_id: measurementDefinition.taxon_measurement_id,
+              measurement_value: row[measurementDefinition.taxon_measurement_id]
+            });
+          }
+        }
+      }
 
-  const isSaving: boolean = _isSavingData.current || _isStoppingEdit.current;
+      // Return the qualitative and quantitative arrays
+      return { qualitative, quantitative };
+    },
+    [measurementColumns]
+  );
+
+  /**
+   * Compiles all subcount columns and their values for the given row.
+   *
+   * @param {ObservationRecord} row
+   * @return {*}
+   */
+  const _getSubcountsToSave = useCallback(
+    (row: ObservationRecord) => {
+      // Get all populated measurement column values for the row
+      const measurementsToSave = _getMeasurementsToSave(row);
+
+      // Return the subcount row to save
+      return {
+        observation_subcount_id: row.observation_subcount_id,
+        // TODO - 1 subcount per observation: Currently the observations datagrid has no subcount column, use the count
```

Comment:
This to-do could use some clarification on why it's there

Reply:
same goes for the one beneath it

Reply:
Adding more detail and a link to a jira ticket